docs/development/release.md

```diff
@@ -3,148 +3,297 @@
 
-* Announce the `rc0` release on the KubeRay slack, with deployment instructions ([example](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1680555251566609)).
+1. **Create Release Branch (if it doesn't exist):**
+    * Clone the `kuberay-helm` repository if you haven't already.
+    * Set up an `upstream` remote: `git remote add upstream git@github.com:ray-project/kuberay-helm.git`
+    * If the branch `release-1.4` does *not* exist in `kuberay-helm`:
 
-#### Step 4. Create more release candidates (`rc1`, `rc2`, ...) if necessary
+        ```bash
+        cd kuberay-helm
+        git fetch upstream
+        # Create the branch from the latest main branch
+        git checkout -b release-1.4 upstream/main
+        git push upstream release-1.4
+        # Set tracking branch
+        git branch --set-upstream-to upstream/release-1.4
+        cd ..
+        ```
 
-* Resolve issues with the release branch by cherry-picking master commits into the release branch.
-* When cherry-picking changes, it is best to open a PR against the release branch -- don't push directly to the release branch.
-* Follow step 3 to create new Docker images and Helm charts for the new release candidate.
+    * *Note:* Creating a new release branch might trigger CI in `kuberay-helm`, but it shouldn't publish new chart versions yet as the chart files haven't changed.
 
-#### Step 5. Create a final release
+2. **Sync Helm Charts:**
+    * Ensure your local `kuberay` repository is on the release branch (`release-1.4`) and up-to-date (including the version bump from Step 2).
 
-* Create a final release (i.e. v0.5.0) by repeating step 4 once more using the tag of the release (`vX.Y.Z`) with no `-rc` suffix.
+        ```bash
+        cd kuberay
+        git checkout release-1.4
+        git pull --rebase upstream release-1.4
+        cd ..
+        ```
 
-#### Step 6. Merge open PRs in step 1 and post-release PRs
+    * Copy the updated Helm charts from the `kuberay` repo to the `kuberay-helm` repo. *This assumes `kuberay` and `kuberay-helm` are in the same parent directory.*
 
-Now, we have the Docker images and Helm charts for v0.5.0.
+        ```bash
+        cd kuberay-helm
+        # Ensure you are on the correct release branch
+        git checkout release-1.4
+        git pull --rebase upstream release-1.4
 
-* Merge the pull requests in Step 1 (i.e. [#997](https://github.com/ray-project/kuberay/pull/997), [#999](https://github.com/ray-project/kuberay/pull/999), [#1004](https://github.com/ray-project/kuberay/pull/1004), [#1012](https://github.com/ray-project/kuberay/pull/1012))
+        # Remove old charts first to handle deleted files
+        rm -rf helm-chart/
+        # Copy the updated charts from the kuberay repo
+        cp -R ../kuberay/helm-chart/ .
+        # Remove test file not needed in kuberay-helm repo
+        # (Adjust path if needed)
+        rm -f helm-chart/script/rbac_test.py
```

Comment:
Maybe we can move this to another directory, so we don't need to manually delete it.

Reply:
open a good first issue: https://github.com/ray-project/kuberay/issues/3255.

This PR doesn't need to be blocked by the issue.

Reply:
The PR has already been merged, so `rm -f` command can be removed.